### PR TITLE
Update README to mention Windows support.

### PR DESCRIPTION
**Pull request by hawkinsp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -413,9 +410,16 @@ pip install --upgrade "jax[cpu]"
...
-[above](#installation).**
+`manylinux2014` wheels. We currently release `jaxlib` wheels for the following
+operating systems and architectures:
+* Linux, x86-64
+* Mac (Intel)
+* Mac (ARM)
+* Windows, x86-64 (*experimental*)
+
+Other architectures require building from source. Trying to pip install on
+other architectures may lead to `jaxlib` not being installed alongside `jax`,
```
Comment (mattjj) [inline README.md:421]:
```suggestion
Other platforms require building from source. Trying to pip install on
other platforms may lead to `jaxlib` not being installed alongside `jax`,
```

I'm guessing platform ≈ OS + architecture.

**Comment by rasswanth-s** [comment]
Thank you @hawkinsp  for the windows wheels release, 
I have been following the windows release thread for some time.
Glad that it is finally released.

I am a contributor at https://github.com/OpenMined/PySyft

This would greatly help us.

File: README.md (hunk 1 above)
Comment (skye) [reply README.md:421]: FWIW I've been using "platform" to mean accelerator architecture, so I kinda like architecture to distinguish (not that it's unclear in this context) and I do think architecture is more precise.

**Review by skye — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (mattjj) [reply README.md:421]: Being consistent with your terminology (which may be standard) SGTM, but my suggestion was just an attempt to make this text consistent with the text just above, especially "We currently release `jaxlib` wheels for the following operating systems and architectures". Based on that sentence I took architecture to mean something like hardware/ISA.

Comment (hawkinsp) [reply README.md]: Changed to "operating systems and architectures" in this paragraph.

**Pull request merged**